lib/src/clojure_lsp/crawler.clj

```diff
@@ -26,56 +26,16 @@
 (defn ^:private analyze-source-paths! [paths db*]
   (let [result (shared/logging-time
                  (str startup-logger-tag " Project only paths analyzed, took %s")
-                 (lsp.kondo/run-kondo-on-paths! paths false db*))
-        analysis (->> (:analysis result)
-                      (lsp.kondo/normalize-analysis false)
-                      (group-by :filename))]
-    (swap! db* (fn [state-db]
-                 (-> state-db
-                     (update :analysis merge analysis)
-                     (assoc :kondo-config (:config result))
-                     (update :findings merge (group-by :filename (:findings result))))))
-    analysis))
-
-(defn ^:private report-batch-analysis-percentage
-  [start-progress-percentage fulfill-progress-percentage progress-token producer index count]
-  (let [real-percentage (- fulfill-progress-percentage start-progress-percentage)]
-    (producer/publish-progress
-      producer
-      (+ start-progress-percentage (/ (* index real-percentage) count))
-      "Analyzing external classpath"
-      progress-token)))
-
-(defn ^:private analyze-external-classpath!
-  [paths start-progress-percentage fulfill-progress-percentage progress-token {:keys [db* producer]}]
-  (let [batch-update-callback (partial report-batch-analysis-percentage start-progress-percentage fulfill-progress-percentage progress-token producer)
-        result (shared/logging-time
-                 "External classpath paths analyzed, took %s. Caching for next startups..."
-                 (lsp.kondo/run-kondo-on-paths-batch! paths true batch-update-callback db*))
-        kondo-analysis (-> (:analysis result)
-                           (dissoc :namespace-usages :var-usages)
-                           (update :var-definitions (fn [usages] (remove :private usages))))
-        analysis (->> kondo-analysis
-                      (lsp.kondo/normalize-analysis true)
-                      (group-by :filename))]
-    (swap! db* assoc :kondo-config (:config result))
-    analysis))
+                 (lsp.kondo/run-kondo-on-paths! paths db* {:external? false}))]
+    (swap! db* db/merge-kondo-results result)))
 
 (defn analyze-reference-filenames! [filenames db*]
-  (let [result (lsp.kondo/run-kondo-on-reference-filenames! filenames db*)
-        analysis (->> (:analysis result)
-                      (lsp.kondo/normalize-analysis false)
-                      (group-by :filename))
-        empty-findings (reduce (fn [map filename] (assoc map filename [])) {} filenames)
-        new-findings (merge empty-findings (group-by :filename (:findings result)))]
-    (swap! db* (fn [db]
-                 (-> db
-                     (update :analysis merge analysis)
-                     (assoc :kondo-config (:config result))
-                     (update :findings merge new-findings))))
-    analysis))
+  (let [result (lsp.kondo/run-kondo-on-reference-filenames! filenames db*)]
+    (swap! db* db/merge-kondo-results result)))
```

Comment:
I still have the feeling `merge-kondo-results` is something that could be on kondo ns, not that related with db but the result map we would create from kondo analysis?

Reply:
I'll work on moving `merge-kondo-results` over to the kondo ns. Will let you know when that's done.